pkg/tests/apis/alerting/notifications/receivers/receiver_test.go

```diff
@@ -0,0 +1,927 @@
+			Spec: v0alpha1.ReceiverSpec{
+				Title:        defaultReceiver.Spec.Title,
+				Integrations: nil,
+			},
+		}
+		_, err := adminClient.Create(ctx, newReceiver, v1.CreateOptions{})
+		require.Truef(t, errors.IsBadRequest(err), "Expected BadRequest, got: %s", err)
+	})
+
+	t.Run("should not let delete default receiver", func(t *testing.T) {
+		err := adminClient.Delete(ctx, defaultReceiver.Name, v1.DeleteOptions{})
+		require.Truef(t, errors.IsConflict(err), "Expected Conflict, got: %s", err)
+	})
+
+	var receiver *v0alpha1.Receiver
+	t.Run("should correctly persist all known integrations", func(t *testing.T) {
+		integrations := make([]v0alpha1.Integration, 0, len(notify.AllKnownConfigsForTesting))
+		keysIter := maps.Keys(notify.AllKnownConfigsForTesting)
+		keys := slices.Collect(keysIter)
+		sort.Strings(keys)
+		for _, key := range keys {
+			integrations = append(integrations, createIntegration(t, key))
+		}
+
+		receiver, err = adminClient.Create(ctx, &v0alpha1.Receiver{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: "default",
+			},
+			Spec: v0alpha1.ReceiverSpec{
+				Title:        "all-receivers",
+				Integrations: integrations,
+			},
+		}, v1.CreateOptions{})
+		require.NoError(t, err)
+		require.Len(t, receiver.Spec.Integrations, len(integrations))
+
+		// Use export endpoint because it's the only way to get decrypted secrets fast.
+		cliCfg := helper.Org1.Admin.NewRestConfig()
+		legacyCli := alerting.NewAlertingLegacyAPIClient(helper.GetEnv().Server.HTTPServer.Listener.Addr().String(), cliCfg.Username, cliCfg.Password)
+
+		export := legacyCli.ExportReceiverTyped(t, receiver.Spec.Title, true)
+		for _, integration := range export.Receivers {
+			expected := notify.AllKnownConfigsForTesting[strings.ToLower(integration.Type)] // to lower because there is LINE that is in different casing in API
+			assert.JSONEqf(t, expected.Config, string(integration.Settings), "integration %s", integration.Type)
+		}
+	})
+
+	t.Run("should be able read what it is created", func(t *testing.T) {
+		get, err := adminClient.Get(ctx, receiver.Name, v1.GetOptions{})
+		require.NoError(t, err)
+		require.Equal(t, receiver, get)
+		t.Run("should return secrets in secureFields but not settings", func(t *testing.T) {
+			for _, integration := range get.Spec.Integrations {
+				t.Run(integration.Type, func(t *testing.T) {
+					secretFields, err := channels_config.GetSecretKeysForContactPointType(integration.Type)
+					require.NoError(t, err)
+					for _, field := range secretFields {
+						assert.Contains(t, integration.SecureFields, field)
+						assert.Truef(t, integration.SecureFields[field], "secure field should be always true")
+
+						value, ok, err := unstructured.NestedString(integration.Settings.Object, strings.Split(field, ".")...)
+						assert.NoErrorf(t, err, "failed to read field %s from settings", field)
+						assert.Falsef(t, ok, "secret field %s should not be in settings, value [%s]", field, value)
+					}
+				})
+			}
+		})
+	})
```

Comment:
Snippet mentioned above to cover invalid integrations, feel free to modify as you wish.
```suggestion
	})

	t.Run("should fail to persist receiver with invalid config", func(t *testing.T) {
		keysIter := maps.Keys(notify.AllKnownConfigsForTesting)
		keys := slices.Collect(keysIter)
		sort.Strings(keys)
		for _, key := range keys {
			t.Run(key, func(t *testing.T) {
				integration := createIntegration(t, key)
				// Make the integration invalid, so it fails to create. This is usually done by sending empty settings.
				clear(integration.Settings.Object)
				if key == "webex" {
					// Webex integration is special case and passes validation without any settings so we instead set an invalid URL.
					integration.Settings.Set("api_url", "(*^$*^%!@#$*()")
				}

				receiver, err = adminClient.Create(ctx, &v0alpha1.Receiver{
					ObjectMeta: v1.ObjectMeta{
						Namespace: "default",
					},
					Spec: v0alpha1.ReceiverSpec{
						Title:        fmt.Sprintf("invalid-%s", key),
						Integrations: []v0alpha1.Integration{integration},
					},
				}, v1.CreateOptions{})
				require.Errorf(t, err, "Expected error but got successful result: %v", receiver)
				require.Truef(t, errors.IsBadRequest(err), "Expected BadRequest, got: %s", err)
			})
		}
	})
```